test-data/unit/check-namedtuple.test

```diff
@@ -1407,9 +1407,23 @@ from typing import NamedTuple
 class A(NamedTuple):
     x: int
 
-A(x=0).__replace__(x=1)
+replaced = A(x=0).__replace__(x=1)
+reveal_type(replaced)  # N: Revealed type is "__main__.A"
+
 A(x=0).__replace__(x="asdf")  # E: Argument "x" to "__replace__" of "A" has incompatible type "str"; expected "int"
 A(x=0).__replace__(y=1)  # E: Unexpected keyword argument "y" for "__replace__" of "A"
+
+from typing import TypeVar, Generic
+
+T = TypeVar("T")
+
+class GenericA(NamedTuple, Generic[T]):
+    x: T
+
+replaced_2 = GenericA(x=0).__replace__(x=1)
+reveal_type(replaced_2)  # N: Revealed type is "__main__.GenericA"
+GenericA(x=0).__replace__(x="abc") # E: Argument "x" to "__replace__" of "GenericA" has incompatible type "str"; expected "int"
```

Comment:
It's specialized, but the generic isn't in the revealed type which is odd.

Reply:
I think that's how mypy generally works:

```
% mypy -c 'x: list[str] = []; x.append(1)'
<string>:1: error: Argument 1 to "append" of "list" has incompatible type "int"; expected "str"  [arg-type]
```

Agree this is potentially confusing but it's not something that should be changed in this PR.

Reply:
Ah I see, makes sense - let me bring this out of draft in that case.